src/impls/$select-by/$select-by.js

```diff
@@ -0,0 +1,30 @@
+import { $async, $await } from '../../../generate/async.macro.cjs';
+
+import { $iterableCurry } from '../../internal/$iterable.js';
+import { $__peekerate } from '../$peekerate/$peekerate.js';
+
+$async;
+export function $__selectBy(iterable, selector, mapper) {
+  const peekr = $await($__peekerate(iterable));
+
+  if (!peekr.done) {
+    let bestItem = peekr.value;
+    let bestValue = mapper(bestItem);
```

Comment:
```js
let bestValue = $await(mapper(bestItem));
```

Reply:
Does that mean the mapper function could return a promise in the async implementation ?
I guess I should also update the typings to reflect this ?

With a FP background, I am not too fond of having mapper functions being asynchronous, possibly side-effecting (I think these kind of functions should be pure 99% of time), but it would be more flexible to allow so.

Reply:
Yes, it does. I have chosen to allow that throughout the library. I expect that most users will not need or want to use that feature, but it's there.